### PR TITLE
Add decorator a class

### PR DESCRIPTION
**Pull request by Jeanbulambo**
In this PR, we implemented the following elements :+1: 
- Check the usage of the Decorator pattern in Ruby
- we use two decorators in order to capitalize and trim people's names.
- Create a class Nameable
- Make sure that your Person class inherits from Nameable
- Create a class that inherits from the base Decorator class.
- Create a class that inherits from the base Decorator class

### CODE REVIEW

File: decoration.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+equire_relative './nameable'
```
Comment (Davidosky007) [inline decoration.rb]: * Kindly fix this typo here instead of `require_relative` you have `equire_relative`

File: person.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
+equire_relative './nameable'
```
Comment (Davidosky007) [inline person.rb]: * Same typo here, Kindly correct the `require_relative` spelling here

File: trimmerdecorator.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class TrimmerDecorator < Decorator
```
Comment (Davidosky007) [inline trimmerdecorator.rb:3]: * Since you are inheriting from class `Decorator` kindly require the decorator class in here 👍

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Jeanbulambo 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* Proper use of Gitflow 👍 
* Well written PR description 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @Jeanbulambo, 


68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Good commit messages 👍 
✅ Linters are passing 👏 


## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**